tap_core/tests/manager_test.rs

```diff
@@ -204,6 +205,68 @@ async fn manager_create_rav_request_all_valid_receipts(
         .is_ok());
 }
 
+#[rstest]
+#[tokio::test]
+async fn deny_rav_due_to_wrong_value(
+    keys: (LocalWallet, Address),
+    allocation_ids: Vec<Address>,
+    domain_separator: Eip712Domain,
+    context: ContextFixture,
+) {
+    let ContextFixture {
+        context,
+        checks,
+        query_appraisals,
+        escrow_storage,
+        ..
+    } = context;
+    let manager = Manager::new(domain_separator.clone(), context, checks);
+    escrow_storage.write().unwrap().insert(keys.1, 999999);
+
+    let mut stored_signed_receipts = Vec::new();
+    for _ in 0..10 {
+        let value = 20u128;
+        let signed_receipt = EIP712SignedMessage::new(
+            &domain_separator,
+            Receipt::new(allocation_ids[0], value).unwrap(),
+            &keys.0,
+        )
+        .unwrap();
+        let query_id = signed_receipt.unique_hash();
+        stored_signed_receipts.push(signed_receipt.clone());
+        query_appraisals.write().unwrap().insert(query_id, value);
+        let _ = manager.verify_and_store_receipt(signed_receipt).await;
+    }
+    let rav_request_result = manager.create_rav_request(0, None).await;
+
+    let rav_request = rav_request_result.unwrap();
+
+    // Create more receipts to create a manual rav with different value
+    let mut receipts = Vec::new();
+    for value in 50..60 {
+        receipts.push(
+            EIP712SignedMessage::new(
+                &domain_separator,
+                Receipt::new(allocation_ids[0], value).unwrap(),
+                &keys.0,
+            )
+            .unwrap(),
+        );
+    }
+
+    let signed_rav_with_wrong_aggregate = EIP712SignedMessage::new(
+        &domain_separator,
+        ReceiptAggregateVoucher::aggregate_receipts(allocation_ids[0], &receipts, None).unwrap(),
+        &keys.0,
+    )
+    .unwrap();
+
+    assert!(manager
+        .verify_and_store_rav(rav_request.expected_rav, signed_rav_with_wrong_aggregate)
+        .await
+        .is_err());
```

Comment:
In this test, `rav_request.expected_rav` and `signed_rav_with_wrong_aggregate` actually differ in value *and* timestamp.

If you want to test what happens if only the value differs, you're better off just creating the expected RAV and the received RAV manually here.
So actually when you think about that, turns out you can simplify this test greatly, and skip creating receipts altogether.

Reply:
Oof you are right, thanks for noticing that